ddtrace/appsec/_iast/_taint_tracking/Aspects/AspectStr.cpp

```diff
@@ -0,0 +1,116 @@
+#include <Aspects/AspectStr.h>
+
+py::str
+api_str_aspect(const py::object& orig_function,
+               const int flag_added_args,
+               const py::args& args,
+               const py::kwargs& kwargs)
+{
+    auto result_or_args = py::reinterpret_borrow<py::object>(
+      process_flag_added_args(orig_function.ptr(), flag_added_args, args.ptr(), kwargs.ptr()));
+
+    py::tuple args_tuple;
+    if (py::isinstance<py::tuple>(result_or_args)) {
+        args_tuple = result_or_args.cast<py::tuple>();
+    } else {
+        return result_or_args;
+    }
+
+    const py::object text = args_tuple[0];
+    const py::str encoding = len(args) > 1 ? args_tuple[1] : py::str("");
+    const py::str errors = len(args) > 2 ? args_tuple[2] : py::str("");
+
+    py::str result_o;
+
+    // Call the original if not a text type
+    if (not is_text(text.ptr())) {
+        PyObject* as_str = PyObject_Str(text.ptr());
+        if (as_str == nullptr) {
+            throw py::error_already_set();
+        }
+        return py::reinterpret_borrow<py::str>(as_str);
+    }
+
+    // With no encoding or errors we can also directly call PyObject_Str
+    if (len(args_tuple) == 1 or (len(encoding) == 0)) {
+        PyObject* as_str = PyObject_Str(text.ptr());
+        if (as_str == nullptr) {
+            throw py::error_already_set();
+        }
+        result_o = py::reinterpret_borrow<py::str>(as_str);
+    } else {
+        // Bytesomething: we have to decode
+
+        // If it has encoding, then the text object must not be a unicode object
+        if (len(encoding) > 0 and py::isinstance<py::str>(text)) {
+            PyErr_SetString(PyExc_TypeError, "decoding str is not supported");
+            throw py::error_already_set();
+        }
+
+        const char* char_encoding = encoding.cast<string>().c_str();
+        const char* char_errors = errors.cast<string>().c_str();
+
+        char* text_raw_bytes;
+        Py_ssize_t text_raw_bytes_size;
+        if (PyBytes_AsStringAndSize(text.ptr(), &text_raw_bytes, &text_raw_bytes_size) == -1) {
+            throw py::error_already_set();
+        }
+
+        PyObject* result_pyo = PyUnicode_Decode(text_raw_bytes, text_raw_bytes_size, char_encoding, char_errors);
+        if (PyErr_Occurred()) {
+            throw py::error_already_set();
+        }
+        if (result_pyo == nullptr) {
+            return py::none();
+        }
+        result_o = py::reinterpret_borrow<py::str>(result_pyo);
+    }
+
+    TRY_CATCH_ASPECT("str_aspect", return result_o, , {
```

Comment:
Are the two commas (`, ,`) intended? What does this do?

Reply:
Yes it is, you can check other aspects
https://github.com/DataDog/dd-trace-py/blob/main/ddtrace/appsec/_iast/_taint_tracking/Aspects/AspectIndex.cpp#L81